lib/watir_drops/page_object.rb

```diff
@@ -97,8 +119,27 @@ def fill_form(model)
     def inspect
       '#<%s url=%s title=%s>' % [self.class, url.inspect, title.inspect]
     end
+
     alias selector_string inspect
 
+
+    def on_page?
+      begin
+        Watir::Wait.until { page_url.gsub(/.*:\/\//i, '').gsub(/\/$/i, '') == (@browser.url.gsub(/.*:\/\//i, '').gsub(/\/$/i, '')) } if @require_url
+
+        Watir::Wait.until { @browser.title == page_title } if @require_title
+
+        if self.class.required_element_list.any?
+          Watir::Wait.until { self.class.required_element_list.all? { |e| send(e).present? } }
+        end
+      rescue
+        false
+      else
+        true
```

Comment:
you don't need the else statement. If it doesn't throw an exception it will return the last line, so put the `true` right before the rescue